ci/ReleaseTool/Common.cs

```diff
@@ -55,5 +57,108 @@ private static string AppendDirectorySeparator(string originalPath)
 
             return originalPath + directorySeparator;
         }
+
+        public static bool IsMarkdownHeading(string markdownLine, int level, string startTitle = null)
+        {
+            var heading = $"{new string('#', level)} {startTitle ?? string.Empty}";
+
+            return markdownLine.StartsWith(heading);
+        }
+
+        public static bool UpdateChangeLog(string ChangeLogFilePath, string Version, GitClient gitClient, string ChangeLogReleaseHeadingTemplate)
+        {
+            using (new WorkingDirectoryScope(gitClient.RepositoryPath))
+            {
+                if (File.Exists(ChangeLogFilePath))
+                {
+                    var originalContents = File.ReadAllText(ChangeLogFilePath);
+                    var changelog = File.ReadAllLines(ChangeLogFilePath).ToList();
+                    var releaseHeading = string.Format(ChangeLogReleaseHeadingTemplate, Version,
+                        DateTime.Now);
+                    var releaseIndex = changelog.FindIndex(line => IsMarkdownHeading(line, 2, $"[`{Version}`] - "));
+                    // If we already have a changelog entry for this release, replace it.
+                    if (releaseIndex != -1)
+                    {
+                        changelog[releaseIndex] = releaseHeading;
+                    }
+                    else
+                    {
+                        // Add the new release heading under the "## Unreleased" one.
+                        // Assuming that this is the first heading.
+                        var unreleasedIndex = changelog.FindIndex(line => IsMarkdownHeading(line, 2));
+                        changelog.InsertRange(unreleasedIndex + 1, new[]
+                        {
+                            string.Empty,
+                            releaseHeading
+                        });
+                    }
+                    File.WriteAllLines(ChangeLogFilePath, changelog);
+
+                    // If nothing has changed, return false, so we can react to it from the caller.
+                    if (File.ReadAllText(ChangeLogFilePath) == originalContents)
+                    {
+                        return false;
```

Comment:
Errors?

Reply:
Ah, this one is actually not an error. The file could already be updated, for example when rerunning the step, in which case we want to continue along, but react to it from above, which is why I added this bool return.